components/admin_console/group_settings/group_details/group_details.jsx

```diff
@@ -115,14 +132,59 @@ export default class GroupDetails extends React.PureComponent {
         this.setState({loadingTeamsAndChannels: false});
     }
 
-    onToggle = async (allowReference) => {
-        this.setState({allowReference});
-        this.props.actions.patchGroup(this.props.groupID, {allow_reference: allowReference});
+    onMentionToggle = async (allowReference) => {
+        const {group} = this.props;
+        const isOriginalAllowReference = group.allow_reference;
+
+        this.setState({
+            saveNeeded: true,
+            allowReference,
+            hasAllowReferenceChanged: allowReference !== isOriginalAllowReference}
+        );
+    }
+
+    onMentionChange = async (e) => {
+        const {group} = this.props;
+        const isOriginalGroupMentionName = group.name;
+        const groupMentionName = e.target.value;
+        const saveNeeded = true;
+
+        this.setState({
+            saveNeeded,
+            groupMentionName,
+            hasGroupMentionNameChanged: groupMentionName !== isOriginalGroupMentionName
+        });
+        this.props.actions.setNavigationBlocked(saveNeeded);
+    }
+
+    handleSubmit = async () => {
+        this.setState({saving: true});
+        const {allowReference, groupMentionName, hasAllowReferenceChanged, hasGroupMentionNameChanged} = this.state;
+
+        let serverError = null;
+        let saveNeeded = false;
+
+        if (!groupMentionName || groupMentionName.length === 0) {
```

Comment:
Why do we need a `!groupMentionName` ? If `groupMentionName` is a string, wouldn't the `groupMentionName.length === 0` be enough if your checking for a empty string? You initialize it as a string object so I'm lead to believe it will never be anything but a string. Also, I think it may be a little more clearer if you did `groupMentionName === ""` instead of using `.length` just to make it more definitive that it's a string (but this just for more clarity and isn't something that needs to be changed).

Reply:
Actually, according to https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in just `!groupMentionName` should be enough.

Reply:
I know this is common way to check for falsy values but I personally dislike it 😆 It makes it feel like I'm working with a boolean when I'm not. It just makes it more readable when you do `variable === ''` so then I know immediately this should be a string and not something else. But I'm 0/5 and if you prefer to keep it `!groupMentionName` then I'm fine with it too.